### PR TITLE
JAMES-3946 DropList Cassandra and Memory app

### PR DESCRIPTION
**Pull request by Maxxx873**
[https://issues.apache.org/jira/projects/JAMES/issues/JAMES-3946](https://issues.apache.org/jira/projects/JAMES/issues/JAMES-3946?filter=allopenissues)

### CODE REVIEW

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/index.adoc [hunk 1]
```diff
@@ -52,6 +52,7 @@ By omitting these files, sane default values are used.
 ** xref:configure/recipientrewritetable.adoc[*recipientrewritetable.xml*] enables advanced configuration for the Recipient Rewrite Table component link:https://github.com/apache/james-project/blob/master/server/apps/distributed-app/sample-configuration/recipientrewritetable.xml[example]
 *** xref:configure/matchers.adoc[This page] allows choosing the indexing technology.
 ** xref:configure/usersrepository.adoc[*usersrepository.xml*] allows configuration of user storage link:https://github.com/apache/james-project/blob/master/server/apps/distributed-app/sample-configuration/usersrepository.xml[example]
+** xref:configure/droplists.adoc[This page] allows configuring drop lists.
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/index.adoc:55]: I would move this down into "extensions" as this is not enabled by default this likely isn't "core"

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/droplists.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+It is disabled by default.
+To enable it, modify the `droplists.properties` file and include the `IsInDropList` matcher in the `mailetcontainer.xml`.
+To disable it, adjust the `droplists.properties` file and remove the `IsInDropList` matcher from the `mailetcontainer.xml`.
+
+.droplists.properties content
+|===
+| Property name | explanation
+
+| enabled
+| true/false. Governs whether DropLists should be enabled
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/droplists.adoc:15]: Please indicate the default value. And the fact that we expect a `Boolean`.

File: server/apps/distributed-app/docs/modules/ROOT/pages/configure/droplists.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+= Distributed James Server &mdash; DropLists
+:navtitle: DropLists
+
+The DropList, also known as the mail blacklist, is a collection of
+domains and email addresses that are denied from sending emails within the system.
+It is disabled by default.
+To enable it, modify the `droplists.properties` file and include the `IsInDropList` matcher in the `mailetcontainer.xml`.
+To disable it, adjust the `droplists.properties` file and remove the `IsInDropList` matcher from the `mailetcontainer.xml`.
```
Comment (chibenwa) [inline server/apps/distributed-app/docs/modules/ROOT/pages/configure/droplists.adoc:8]: Please indicate me how I can add entries onto my bloclist!

**Review by chibenwa — COMMENTED** [review]
Looks cool, thanks!

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Hello @Maxxx873 

I think that with this PR we are feature complete.

I would suggest to merge the `droplist` branch onto `master` and do the complementary developments against the master branch.

Thoughts? Do you want to contribute this merge into master?

Thanks and best regards.

**Comment by Maxxx873** [comment]
> Hello @Maxxx873
> 
> I think that with this PR we are feature complete.
> 
> I would suggest to merge the `droplist` branch onto `master` and do the complementary developments against the master branch.
> 
> Thoughts? Do you want to contribute this merge into master?
> 
> Thanks and best regards.



> Hello @Maxxx873
> 
> I think that with this PR we are feature complete.
> 
> I would suggest to merge the `droplist` branch onto `master` and do the complementary developments against the master branch.
> 
> Thoughts? Do you want to contribute this merge into master?
> 
> Thanks and best regards.

Hello Benoit. I will be glad if you merge the "droplist" with the "master" branch. After the merge, I am planning some additions - a JPA backend, as an alternative to `AbstractSQLWhitelistMatcher`, `IsInWhiteList`, `WhiteListManager`. Also, I would be interested in implementing a postgres backend as well. I appreciate any suggestions and advice.

**Pull request merged**